store/src/main/java/org/apache/rocketmq/store/MultiPathMappedFileQueue.java

```diff
@@ -62,33 +70,75 @@ public boolean load() {
     @Override
     protected MappedFile tryCreateMappedFile(long createOffset) {
         long fileIdx = createOffset / this.mappedFileSize;
-        List<String> pathList = config.getCommitLogStorePaths();
+        List<String> pathList = checkDiskSpaceAndReturnPaths();
+        if (pathList.size() <= 0) {
+            return null;
+        }
         String nextFilePath = pathList.get((int) (fileIdx % pathList.size())) + File.separator
                 + UtilAll.offset2FileName(createOffset);
         String nextNextFilePath = pathList.get((int) ((fileIdx + 1) % pathList.size())) + File.separator
                 + UtilAll.offset2FileName(createOffset + this.mappedFileSize);
         return doCreateMappedFile(nextFilePath, nextNextFilePath);
     }
 
+    private List<String> checkDiskSpaceAndReturnPaths() {
+        List<String> storePaths = new ArrayList<>();
+        String[] checkPaths = getCommitLogStorePaths();
+        if (checkPaths != null) {
+            for (int i = 0;i < checkPaths.length;i++) {
+                String path = checkPaths[i];
+                try {
+                    File file = new File(path);
+                    MappedFile.ensureDirOK(file.getPath());
+                    if (file.exists() && file.getFreeSpace() > this.mappedFileSize) {
+                        storePaths.add(path);
+                    }
+                } catch (Exception e) {
+                    log.error("Failed to load the store path{} ex{}",path,e.getMessage());
+                }
+            }
+        }
+
+        return storePaths;
+    }
+
+    public String[] getCommitLogStorePaths() {
+        String[] storPaths = null;
+        if (!UtilAll.isBlank(config.getCommitLogStorePaths())) {
+            storPaths = config.getCommitLogStorePaths().trim().split(";") ;
```

Comment:
';' is really not a shell friendly file path separator. IMHO,  ','  or ':' is more commonly used.